wgengine/magicsock/magicsock.go

```diff
@@ -2132,64 +2132,84 @@ func (c *Conn) handlePingLocked(dm *disco.Ping, src epAddr, di *discoInfo, derpN
 	di.lastPingTime = time.Now()
 	isDerp := src.ap.Addr() == tailcfg.DerpMagicIPAddr
 
-	if src.vni.isSet() {
-		// TODO(jwhited): check for matching [endpoint.bestAddr] once that data
-		//  structure is VNI-aware and [relayManager] can mutate it. We do not
-		//  need to reference any [endpointState] for Geneve-encapsulated disco,
-		//  we store nothing about them there.
-		c.relayManager.handleGeneveEncapDiscoMsgNotBestAddr(dm, di, src)
-		return
-	}
-
-	// If we can figure out with certainty which node key this disco
-	// message is for, eagerly update our IP:port<>node and disco<>node
-	// mappings to make p2p path discovery faster in simple
-	// cases. Without this, disco would still work, but would be
-	// reliant on DERP call-me-maybe to establish the disco<>node
-	// mapping, and on subsequent disco handlePongConnLocked to establish
-	// the IP:port<>disco mapping.
-	if nk, ok := c.unambiguousNodeKeyOfPingLocked(dm, di.discoKey, derpNodeSrc); ok {
-		if !isDerp {
-			c.peerMap.setNodeKeyForEpAddr(src, nk)
-		}
-	}
-
+	var numNodes int
```

Comment:
Maybe: add a comment explaining numNodes? To someone unfamiliar with magicsock, it takes a fair bit of digging to find the codepath where we can end up discovering that there are multiple endpoints attached to one discokey, all the other codepaths end up using this as a bool.

This isn't a deficiency of this PR, the issue of having ambiguous disco keys is a longstanding confusing corner of magicsock. This PR doesn't make that issue worse, so also fine to skip or defer this to a later time, to remain focused on the feature change.